### PR TITLE
[ARM-200] fix: user cannot accept or reject its own invitation

### LINKED ISSUE #23: [Server] If I try to accept an invitation that I created, I should get a friendly message with instructions on how to send an invitation to a friend.
Estás tratando de aceptar una invitación que tu creaste.

Armore está diseñado para que compartas tu ubicación con las personas que amas.

Para lograr esto, debes de mandar la invitación (link) a la persona que quieres que te siga.

Si tienes dudas, ve a la sección de perfil y pregúntanos lo que sea por email o discord.

### PR DESCRIPTION
**Pull request by JasterV**
fixes #23

### CODE REVIEW

File: rust/src/lang/english.rs [hunk 1]
```diff
@@ -24,6 +24,7 @@ lazy_static! {
         (TranslationIds::EmergencyModePushNotificationBody, "is in an EMERGENCY! Please CONFIRM that they are okay!!"),
         (TranslationIds::NormalModePushNotificationBody, "is no longer in an emergency."),
         (TranslationIds::InvalidHistoricalLocationStartTime, "It is not possible to obtain the location from more than a week ago."),
-        (TranslationIds::PushNotificationActionView, "Go to app")
+        (TranslationIds::PushNotificationActionView, "Go to app"),
+        (TranslationIds::CannotUseOwnInvitation, "You are dealing with an invitation that you created.\nArmore is designed for you to share your location with the people you love.\nTo achieve this, you must send the invitation (link) to the person you want to follow you.\nIf you have any questions, go to the profile section and ask us anything by email or discord.")
```
Comment (darioalessandro) [inline rust/src/lang/english.rs]: Hey @JasterV , it is more friendly if the first sentence is: You are trying to accept an invitation that you created

File: rust/src/lang/spanish.rs [hunk 2]
```diff
@@ -24,6 +24,7 @@ lazy_static! {
         (TranslationIds::EmergencyModePushNotificationBody, "está en una EMERGENCIA! Por favor CONFIRME que están bien!!"),
         (TranslationIds::NormalModePushNotificationBody, "ya no está en una emergencia."),
         (TranslationIds::InvalidHistoricalLocationStartTime, "No es posible obtener la localización de hace más de una semana"),
-        (TranslationIds::PushNotificationActionView, "Ir a la app")
+        (TranslationIds::PushNotificationActionView, "Ir a la app"),
+        (TranslationIds::CannotUseOwnInvitation, "Estás tratando de usar una invitación que tu creaste.\nArmore está diseñado para que compartas tu ubicación con las personas que amas.\nPara lograr esto, debes de mandar la invitación (link) a la persona que quieres que te siga.\nSi tienes dudas, ve a la sección de perfil y pregúntanos lo que sea por email o discord."),
```
Comment (darioalessandro) [inline rust/src/lang/spanish.rs:28]: Perfect :)

File: rust/src/server/validators/invitations.rs [hunk 3]
```diff
@@ -11,7 +11,20 @@ pub fn assert_valid_invitation(
     conn: &mut PostgresConnection,
     data: &LinkActionData,
 ) -> Result<(), APIInternalError> {
-    get_invitation(conn, &data.uuid).and_then(|row| assert_link_state(conn, &row))
+    get_invitation(conn, &data.uuid).and_then(|row| {
+        assert_invitation_creator(&row, &data.username).and_then(|_| assert_link_state(conn, &row))
+    })
+}
+
+fn assert_invitation_creator(row: &Row, username: &str) -> Result<(), APIInternalError> {
```
Comment (darioalessandro) [inline rust/src/server/validators/invitations.rs:19]: very clean :)

**Review by darioalessandro — CHANGES_REQUESTED** [review]
Hey @JasterV , it is more friendly if the first sentence is: You are trying to accept an invitation that you created

**Comment by JasterV** [comment]
> Hey @JasterV , it is more friendly if the first sentence is: You are trying to accept an invitation that you created
> 
> 

@darioalessandro I've changed it to be more generic so it works for rejecting too

File: rust/src/lang/english.rs (hunk 1 above)
Comment (JasterV) [reply rust/src/lang/english.rs]: For now I'll go for "You are using an invitation that you created" so I can use the same for accept and reject, what do you think?

Comment (JasterV) [reply rust/src/lang/english.rs]: @darioalessandro I've changed it to: "You are using an invitation that you've created", how about that? :D

Comment (griffobeid) [reply rust/src/lang/english.rs]: @JasterV I think that it should be more specific, something like: "You are using an invitation that you created. Send this link to a loved one for them to befriend you."

**Review by darioalessandro — APPROVED** [review]

**Pull request merged**